src/backend/marsha/core/views.py

```diff
@@ -397,6 +408,183 @@ class DocumentView(BaseLTIView):
     serializer_class = DocumentSerializer
 
 
+@method_decorator(csrf_exempt, name="dispatch")
+@method_decorator(xframe_options_exempt, name="dispatch")
+class LTIRespondView(TemplateResponseMixin, View):
+    """LTI view called to respond to a consumer.
+
+    ie. after a deep linking content selection.
+
+    """
+
+    template_name = "core/form_autosubmit.html"
+
+    # pylint: disable=unused-argument
+    def post(self, request, *args, **kwargs):
+        """Respond to POST request.
+
+        Renders a form autosubmitted to a LTI consumer with signed parameters.
+
+        Parameters
+        ----------
+        request : Request
+            passed by Django
+        args : list
+            positional extra arguments
+        kwargs : dictionary
+            keyword extra arguments
+        Returns
+        -------
+        HTML
+            generated from applying the data to the template
+
+        """
+        lti_response_url = self.request.POST.get("lti_response_url")
+
+        # filters out oauth parameters
+        lti_parameters = {
+            key: value
+            for (key, value) in self.request.POST.items()
+            if "oauth" not in key and key not in ("lti_response_url",)
+        }
+
+        # generate signature
+        lti = LTI(self.request)
+        lti_parameters = lti.sign_post_request(lti_response_url, lti_parameters)
+
+        return self.render_to_response(
+            {"form_action": lti_response_url, "form_data": lti_parameters}
+        )
+
+
+@method_decorator(csrf_exempt, name="dispatch")
+@method_decorator(xframe_options_exempt, name="dispatch")
+class LTISelectView(TemplateResponseMixin, View):
+    """LTI view called to select LTI content through Deep Linking.
+
+    It is designed to work as a React single page application.
+
+    """
+
+    template_name = "core/resource.html"
+
+    def get_context_data(self):
+        """Build context for template rendering of configuration data for the frontend.
+
+        Returns
+        -------
+        dictionary
+            context with configuration data for the frontend
+
+        """
+
+        def _manage_exception(error):
+            logger.warning(str(error))
+            return {"state": "error"}
+
+        lti = LTI(self.request)
+
+        if not lti.is_instructor and not lti.is_admin:
+            raise PermissionDenied
+
+        try:
+            lti.verify()
+            Playlist.objects.get_or_create(
+                lti_id=lti.context_id,
+                consumer_site=lti.get_consumer_site(),
+                defaults={"title": lti.context_title},
+            )
+        except LTIException as error:
+            app_data = _manage_exception(error)
+        else:
+            app_data = self._get_app_data(lti)
```

Comment:
I think the playlist query is not concerned by the try and should be placed out of it:
```suggestion
        try:
            lti.verify()
        except LTIException as error:
            app_data = _manage_exception(error)
        else:
            Playlist.objects.get_or_create(
                lti_id=lti.context_id,
                consumer_site=lti.get_consumer_site(),
                defaults={"title": lti.context_title},
            )
            app_data = self._get_app_data(lti)
```